.travis.yml

```diff
@@ -27,8 +27,40 @@ jobs:
         - tests/scripts/all.sh -k build_arm_linux_gnueabi_gcc_arm5vte build_arm_none_eabi_gcc_m0plus
 
     - name: full configuration
+      os: linux
+      dist: focal
+      addons:
+        apt:
+          packages:
+          - clang-10
+          - gnutls-bin
       script:
-        - tests/scripts/all.sh -k test_full_cmake_gcc_asan
+        # Do a manual build+test sequence rather than using all.sh,
+        # because there's no all.sh component that does what we want,
+        # which is a build with Clang >= 10 and ASan, running all the SSL
+        # testing.
+        #   - The clang executable in the default PATH is Clang 7 on
+        #     Travis's focal instances, but we want Clang >= 10.
+        #   - Running all the SSL testing requires a specific set of
+        #     OpenSSL and GnuTLS versions and we don't want to bother
+        #     with those on Travis.
+        # So we explicitly select clang-10 as the compiler, and we
+        # have ad hoc restrictions on SSL testing based on what is
+        # passing at the time of writing. We will remove these limitations
+        # gradually.
+        - make generated_files
+        - make CC=clang-10 CFLAGS='-Werror -Wall -Wextra -fsanitize=address,undefined -fno-sanitize-recover=all -O2' LDFLAGS='-Werror -Wall -Wextra -fsanitize=address,undefined -fno-sanitize-recover=all'
+        - make test
+        - programs/test/selftest
+        - tests/scripts/test_psa_constant_names.py
+        - tests/ssl-opt.sh
```

Comment:
On Travis, "Fallback SCSV: beginning of list", "Fallback SCSV: end of list" and "Fallback SCSV: not in list", the three test cases that use `tests/scripts/tcp_client.pl`, are failing because the client can't connect:
```
Cannot connect to localhost:11192: Connection refused at scripts/tcp_client.pl line 41.
```
I don't understand. This should be a straightforward TCP connection. It's working on my machine.

Reply:
On Travis, this is working on xenial and jammy, but failing on bionic and focal. https://app.travis-ci.com/github/Mbed-TLS/mbedtls/builds/258172893

Reply:
I'm just going to skip these test cases and document the issue.